cms/bundles/tests/test_management_commands.py

```diff
@@ -0,0 +1,205 @@
+        self.bundle.release_calendar_page = release_page
+        self.bundle.save(update_fields=["publication_date", "release_calendar_page"])
+
+        self.call_command()
+
+        # Check release calendar was updated
+        release_page.refresh_from_db()
+        self.assertEqual(release_page.status, ReleaseStatus.PUBLISHED)
+
+        content = release_page.content[0].value
+        self.assertEqual(content["title"], "Publications")
+        self.assertEqual(len(content["links"]), 1)
+        self.assertEqual(content["links"][0]["page"].pk, self.analysis_page.pk)
+
+    @override_settings(SLACK_NOTIFICATIONS_WEBHOOK_URL="https://slack.ons.gov.uk")
+    @patch("cms.bundles.management.commands.publish_bundles.logger")
+    def test_publish_bundle_error_handling(self, mock_logger):
+        """Test error handling during bundle publication."""
+        BundlePageFactory(parent=self.bundle, page=self.analysis_page)
+
+        # Mock an error during publication
+        with patch(
+            "cms.bundles.management.commands.publish_bundles.notify_slack_of_publication_start",
+            side_effect=Exception("Test error"),
+        ):
+            self.call_command()
+
+        # Check error was logged
+        mock_logger.exception.assert_called_with("Publish failed bundle=%d", self.bundle.id)
+
+        # Check bundle status wasn't changed due to error
+        self.bundle.refresh_from_db()
+        self.assertEqual(self.bundle.status, BundleStatus.APPROVED)
+
+    @override_settings(WAGTAILADMIN_BASE_URL="https://test.ons.gov.uk")
+    @override_settings(SLACK_NOTIFICATIONS_WEBHOOK_URL="https://slack.ons.gov.uk")
+    @patch("cms.bundles.management.commands.publish_bundles.notify_slack_of_publication_start")
+    def test_publish_bundle_with_base_url(self, mock_notify):
+        """Test publishing with a configured base URL."""
+        self.call_command()
+
+        # Verify notification was called with correct URL
+        mock_notify.assert_called_once()
+        call_kwargs = mock_notify.call_args[1]
+
+        self.assertEqual(
+            call_kwargs["url"], "https://test.ons.gov.uk" + reverse("bundle:inspect", args=(self.bundle.pk,))
+        )
+        self.assertIn(str(self.bundle.pk), call_kwargs["url"])
+
+
+class PublishScheduledWithoutBundlesCommandTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.home = HomePage.objects.first()
+        cls.analysis_page = AnalysisPageFactory(title="Test Analysis", live=False)
+        cls.bundle = BundleFactory(name="Test Bundle", bundled_pages=[cls.analysis_page])
+
+        cls.publication_date = timezone.now() - timedelta(minutes=1)
+        cls.analysis_page.save_revision(approved_go_live_at=cls.publication_date)
+
+    def setUp(self):
+        self.stdout = StringIO()
+        self.stderr = StringIO()
+
+    def call_command(self, *args, **kwargs):
+        """Helper to call the management command."""
+        call_command(
```

Comment:
Thought: Worth adding some `assertNumQueries` in here to catch performance regressions?

Reply:
good suggestion. One for follow up